contracts/contracts/lib/cometbft/CometbftLightClient.sol

```diff
@@ -0,0 +1,231 @@
+    /// @param i The index of the validator in the active validator set
+    ///
+    /// @return power The voting power of the validated validator
+    /// @return validator The Ethereum address of the validator
+    function getValidatorInfo(uint256 i) internal view returns (uint256 power, address validator) {
+        validator = LibPower.getActiveValidatorAddressByIndex(uint16(i));
+        power = LibPower.getCurrentPower(validator);
+    }
+
+    /// Converts the CanonicalVote.Data into protobuf encoded data required by conmetbft signature verification.
+    function generateSignedPayload(
+        CanonicalVote.Data memory vote
+    ) internal pure returns (bytes memory) {
+        return Encoder.encodeDelim(CanonicalVote.encode(vote));
+    }
+
+    /**
+     * @dev verifies the secp256k1 signature against the public key and message
+     * Tendermint uses RFC6979 and BIP0062 standard, meaning there is no recovery bit ("v" argument) present in the signature.
+     * The "v" argument is required by the ecrecover precompile (https://eips.ethereum.org/EIPS/eip-2098) and it can be either 0 or 1.
+     *
+     * To leverage the ecrecover precompile this method opportunisticly guess the "v" argument. At worst the precompile is called twice,
+     * which still might be cheaper than running the verification in EVM bytecode (as solidity lib)
+     *
+     * See: tendermint/crypto/secp256k1/secp256k1_nocgo.go (Sign, Verify methods)
+     */
+    function verify(bytes32 messageHash, bytes memory signature, address signer) internal pure returns (address recovered, ECDSA.RecoverError err) {
+        (recovered, err) = tryRecover(messageHash, signature, 27);
+        if (err == ECDSA.RecoverError.NoError && recovered != signer) {
+            (recovered, err) = tryRecover(messageHash, signature, 28);
+        }
+    }
+
+    /**
+     * @dev returns the address that signed the hash.
+     * This function flavor forces the "v" parameter instead of trying to derive it from the signature
+     *
+     * Source: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/ECDSA.sol#L56
+     */
+    function tryRecover(bytes32 hash, bytes memory signature, uint8 v) internal pure returns (address recovered, ECDSA.RecoverError err) {
+        if (signature.length == 65 || signature.length == 64) {
+            bytes32 r;
+            bytes32 s;
+            // ecrecover takes the signature parameters, and the only way to get them
+            // currently is to use assembly.
+            assembly {
+                r := mload(add(signature, 0x20))
+                s := mload(add(signature, 0x40))
+            }
+
+            (recovered, err, ) = ECDSA.tryRecover(hash, v, r, s);
+            return (recovered, err);
+        } else {
+            return (address(0), ECDSA.RecoverError.InvalidSignatureLength);
+        }
+    }
+
+    function hashLightHeader(LightHeader.Data memory h) internal pure returns (bytes32) {
```

Comment:
Add docs, including link/ref to the original code, and why it was copied into here.